apps/ejabberd/src/mod_roster.erl

```diff
@@ -424,16 +428,32 @@ do_process_item_set(JID1,
                     To,
                     #xmlel{attrs = Attrs, children = Els}) ->
     LJID = jid:to_lower(JID1),
+    MakeItem2 = fun(Item) ->
+                    Item1 = process_item_attrs(Item, Attrs),
+                    process_item_els(Item1, Els)
+                end,
+    set_roster_item(User, LUser, LServer, LJID, From, To, MakeItem2).
+
+%% @doc this is run when a roster item is to be added, updated or removed
+%% the interface of this func could probably be a bit simpler
+-spec set_roster_item(User :: binary(),
+                      LUser :: binary(),
+                      LServer :: binary(),
+                      LJID :: ejabberd:simple_jid() | error,
+                      From :: jid(),
+                      To :: jid(),
+                      Item2 :: fun( (roster()) -> roster())) -> ok.
+set_roster_item(User, LUser, LServer, LJID, From, To, MakeItem2) ->
     F = fun () ->
-                Item = case mod_roster_backend:get_roster_entry_t(LUser, LServer, LJID) of
+                Item = case get_roster_entry(LUser, LServer, LJID) of
                            does_not_exist ->
                                #roster{usj = {LUser, LServer, LJID},
-                                       us = {LUser, LServer},
-                                       jid = LJID};
+                                   us = {LUser, LServer},
```

Comment:
Indentation got broken here and the line below. Should be indented as previously.